src/library/basetracktablemodel.cpp

```diff
@@ -473,6 +477,41 @@ QVariant BaseTrackTableModel::roleValue(
     case Qt::ToolTipRole:
         if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_COLOR)) {
             return mixxx::RgbColor::toQString(mixxx::RgbColor::fromQVariant(rawValue));
+        } else if (column == fieldIndex(ColumnCache::COLUMN_LIBRARYTABLE_COVERART)) {
+            TrackPointer pTrack = getTrack(index);
```

Comment:
Loading the `Track` object in response to a toolttip trigger is a costly operation. All required properties for `CoverInfo` are available from the table model.

Reply:
The `TrackModel` needs to be extended:
`std::optional<CoverInfo> getCoverInfo(const QModeIndex& index) const`.

Reply:
...then extract all this code into a function.

Reply:
- `std::optional` is not needed, because `CoverInfo` is implicitly optional
- the required code is already part of `CoverArtDelegate` and needs to be pulled into the `TrackModel` implementation. `CoverArtDelegate` should then just invoke `m_pTrackModel->getCoverInfo(index)`.

Reply:
Thanks for the feedback. I wasn't aware of this perfomance implication. I will change this!

Reply:
The `TrackModel` extension and refactoring of `CoverArtDelegate` should ideally be done in a separate PR before we continue with this new feature here.